### PR TITLE
Add support for k8s NetworkPolicy GA

### PR DESCRIPTION
**Pull request by gunjan5**
## Todos
- [x] Clean up old code
- [x] Update comments
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: lib/backend/k8s/conversion_test.go [hunk 1]
```diff
@@ -279,7 +283,10 @@ var _ = Describe("Test NetworkPolicy conversion", func() {
 		Expect(int(*pol.Value.(*model.Policy).Order)).To(Equal(1000))
 		Expect(pol.Value.(*model.Policy).Selector).To(Equal("calico/k8s_ns == 'default' && label == 'value'"))
 		Expect(len(pol.Value.(*model.Policy).InboundRules)).To(Equal(0))
-		Expect(len(pol.Value.(*model.Policy).OutboundRules)).To(Equal(0))
+
+		// Outboundrules should only have one rule and it should be allow.
```
Comment (caseydavenport) [inline lib/backend/k8s/conversion_test.go]: Outboundrules -> OutboundRules (and a few other places)

File: lib/backend/k8s/conversion_test.go [hunk 2]
```diff
@@ -598,15 +595,19 @@ var _ = Describe("Test Namespace conversion", func() {
...
+		Expect(inboundRules[0]).To(Equal(model.Rule{Action: "allow"}))
+		Expect(outboundRules[0]).To(Equal(model.Rule{Action: "allow"}))
 
 		// Check labels.
 		labels := p.Value.(*model.Profile).Labels
 		Expect(len(labels)).To(Equal(0))
 	})
 
-	It("should parse a Namespace to Policy (DefaultDeny)", func() {
+	It("default Profile creation should not fail for the pre-GA namespace isolation annotation (DefaultDeny)", func() {
```
Comment (caseydavenport) [inline lib/backend/k8s/conversion_test.go]: I think this should be of form "should . . ."

So I think this test is actually: "should ignore the network-policy Namespace annotation"

File: lib/backend/k8s/k8s_fv_test.go [hunk 3]
```diff
@@ -209,7 +209,7 @@ var _ = Describe("Test Syncer API for Kubernetes backend", func() {
 		go cb.ProcessUpdates()
 	})
 
-	It("should handle a Namespace with DefaultDeny", func() {
+	It("should handle a Namespace with DefaultDeny (pre-GA annotation for namespace isolation)", func() {
```
Comment (caseydavenport) [inline lib/backend/k8s/k8s_fv_test.go]: Instead of pre-GA, call this `v1beta`

File: lib/backend/k8s/k8s.go [hunk 4]
```diff
@@ -109,14 +109,23 @@ func NewKubeClient(kc *KubeConfig) (*KubeClient, error) {
...
-		&loadingRules, configOverrides).ClientConfig()
+		loadingRules, configOverrides).ClientConfig()
 	if err != nil {
-		return nil, resources.K8sErrorToCalico(err, nil)
+		log.Errorf("*** Error loading config with overrides: %v\n", err)
+		log.Infof("*** config: %v, loadin: %v, configOVerrides: %v", config, loadingRules, configOverrides)
+		//return nil, resources.K8sErrorToCalico(err, nil)
+	}
+
+	config, err = rest.InClusterConfig()
```
Comment (caseydavenport) [inline lib/backend/k8s/k8s.go]: I'm assuming this was just for testing?

Comment (gunjan5) [inline lib/backend/k8s/k8s.go]: yes

File: lib/backend/k8s/k8s.go [hunk 5]
```diff
@@ -105,18 +105,23 @@ func NewKubeClient(kc *KubeConfig) (*KubeClient, error) {
...
 	if err != nil {
+		log.Errorf("*** Error loading config with overrides: %v\n", err)
+		log.Infof("*** config: %v, loadin: %v, configOVerrides: %v", config, loadingRules, configOverrides)
 		return nil, resources.K8sErrorToCalico(err, nil)
 	}
 
 	// Create the clientset
 	cs, err := kubernetes.NewForConfig(config)
 	if err != nil {
+		log.Errorf("** NewForConfig error: %v", err)
```
Comment (caseydavenport) [inline lib/backend/k8s/k8s.go]: Let's clean up / remove these logs

File: lib/backend/k8s/k8s.go [hunk 6]
```diff
@@ -105,18 +105,23 @@ func NewKubeClient(kc *KubeConfig) (*KubeClient, error) {
 		configOverrides.ClusterInfo.InsecureSkipTLSVerify = true
 	}
 	log.Debugf("Config overrides: %+v", configOverrides)
+	log.Infof("*** KC: %v\n", kc)
 
 	// A kubeconfig file was provided.  Use it to load a config, passing through
 	// any overrides.
+	configOverrides = &clientcmd.ConfigOverrides{}
```
Comment (caseydavenport) [inline lib/backend/k8s/k8s.go]: Don't forget to remove this line.

File: lib/backend/k8s/k8s.go [hunk 7]
```diff
@@ -105,18 +105,23 @@ func NewKubeClient(kc *KubeConfig) (*KubeClient, error) {
 		configOverrides.ClusterInfo.InsecureSkipTLSVerify = true
 	}
 	log.Debugf("Config overrides: %+v", configOverrides)
+	log.Infof("*** KC: %v\n", kc)
```
Comment (caseydavenport) [inline lib/backend/k8s/k8s.go]: typo: casey

:p

File: lib/backend/k8s/conversion.go [hunk 8]
```diff
@@ -74,7 +67,6 @@ func (c converter) parsePolicyNameNamespace(name string) (string, error) {
 	}
 
 	return strings.TrimPrefix(name, "ns.projectcalico.org/"), nil
-
```
Comment (gunjan5) [inline lib/backend/k8s/conversion.go]: @caseydavenport  this extra space tho :p

**Comment by gunjan5** [comment]
closing in favor of #441

**Pull request closed without merging**